waterbutler/providers/rushfiles/provider.py

```diff
@@ -95,29 +95,81 @@ def default_headers(self) -> dict:
         return {'authorization': 'Bearer {}'.format(self.token)}
 
     def can_intra_move(self, other: provider.BaseProvider, path: WaterButlerPath=None) -> bool:
-        #TODO check if really possible. Adjust accordingly
         return self == other
 
     def can_intra_copy(self, other: provider.BaseProvider, path=None) -> bool:
-        #TODO check if really possible. Adjust accordingly
-        return self == other
+        # rushfiles can copy a folder, but do not copy the files inside it.
+        return self == other and (path and path.is_file)
 
     async def intra_move(self,  # type: ignore
                          dest_provider: provider.BaseProvider,
                          src_path: WaterButlerPath,
                          dest_path: WaterButlerPath) -> Tuple[BaseRushFilesMetadata, bool]:
-        #TODO remove if can_intra_move is always false.
-        # Check parent implementation and see if it's optimal.
-        # Implement better solution if not, remove override completely if it is.
-        raise NotImplementedError
+
+        src_metadata = await self.metadata(src_path)                 
+        now = self._get_time_for_sending()
+        request_body = json.dumps({
+            'RfVirtualFile': {
+                'ShareId': self.share['id'],
+                'ParrentId': dest_path.parent.identifier,
+                'EndOfFile': src_metadata.size if src_path.is_file else 0,
+                'Tick': 0,
+                'PublicName': dest_path.name,
+                'CreationTime': now,
+                'LastAccessTime': now,
+                'LastWriteTime': now,
+                'Attributes': 32,
+            },
+            'TransmitId': str(self._generate_uuid),
+            'ClientJournalEventType': 16,
+            'DeviceId': 'waterbutler'
+        })
+        
+        async with self.request(
+            'PUT',
+            self._build_filecache_url(str(self.share['id']), 'files', src_path.identifier),
+            data=request_body,
+            headers={'Content-Type': 'application/json'},
+            expects=(200, ),
+            throws=exceptions.IntraMoveError,
+        ) as response:
+            resp = await response.json()
+            data = resp['Data']['ClientJournalEvent']['RfVirtualFile']
+        
+        dest_path.rename(data['PublicName'])
+        
+        if dest_path.is_dir:
+            metadata = RushFilesFolderMetadata(data, dest_path)
+            metadata._children = await self._folder_metadata(dest_path)
```

Comment:
Attributes starting with single underscore should be thought of as private and not accessed directly when possible. `BaseFolderMetadata` has settet/getter pair for this, which would be better to use.

https://github.com/tsukaeru/RDM-waterbutler/blob/develop/waterbutler/core/metadata.py#L384-L400

```suggestion
            metadata.children = await self._folder_metadata(dest_path)
```